rpc/TransactionRPC.java

```diff
@@ -57,7 +57,7 @@
     private final ConceptManager conceptManager;
     private final LogicManager logicManager;
     private final QueryManager queryManager;
-    private final StreamObserver<TransactionProto.Transaction.Req> requestObserver;
+    private final SynchronizedStreamObserver<TransactionProto.Transaction.Req> requestObserver;
```

Comment:
@alexjpwalker note that I didn't replace all StreamObserver usage. For example here on line 180 we have 

`private StreamObserver<TransactionProto.Transaction.Res> responseObserver() {`

Reply:
That should be fine, because each `responseObserver` is only accessed by a single gRPC worker thread!